### PR TITLE
ruby docs generated fully from openapi

### PR DESCRIPTION
**Pull request by SerKnight**
### This PR:
1) Removes legacy (unscalable) doc creation from hardcoding sdk examples
2) Inserts more DRY approach with parameterized values to get ruby docs working

> It is part of our Implementation of our internal roadmap objective defined here: https://confluence.teamxero.com/display/API/SDK+Documentation

## Release Notes
Open API being used to inform the core api docs, and will be used to show language specific docs in future. This also removes the legacy way we generated docs for ruby and node using explicit params/examples

## Screenshots (if appropriate):
Screen Shot 2021-03-24 at 4 54 45 PM

## Types of Changes

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: xero_accounting.yaml [hunk 1]
```diff
@@ -1461,6 +1460,13 @@ paths:
           keyPascal: Contact
           default: contact
           object: bankTransaction
+        - set_lineitems:
+          is_variable: true
+          nonString: true
+          key: lineItems
+          keyPascal: LineItems
+          object: bankTransaction
+          default: lineItems
```
Comment (SerKnight) [inline xero_accounting.yaml:1469]: moving this up seemed to fix the outside of object addition.

```ruby
object = {
  key: "val",
  key2: "val2"
}

  line_items: line_item,
```

**Comment by SerKnight** [comment]
@SidneyAllen  for your review. No rush.

Made comments where I thought necessary.

I will say I don't love the duplication across `ruby` and `python` for certain subsets of snake data.
```yml
  python: bank_account
  ruby: bank_account
```

I did try to consolidate, but because the same pattern is used for actual language specific diffs, like Enums, dates, it make it messier than the duplication.

Lmk any comments/changes.

**Review by AlbertGromek — APPROVED** [review]
Checked over and looks good to me :)

File: xero_accounting.yaml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 openapi: 3.0.0
 info:
   title: Xero Accounting API
-  version: "2.10.3"
+  version: "2.10.2"
```
Comment (AlbertGromek) [inline xero_accounting.yaml:4]: Should this be 2.10.4?

Comment (SerKnight) [reply xero_accounting.yaml:4]: Yes - i've got some dependent changes from a branch off 2.10.2
> https://github.com/XeroAPI/Xero-OpenAPI/pull/391

I'll remerge master on the OAS PR and get this fixed :P

Comment (SerKnight) [reply xero_accounting.yaml:4]: I think this is cached or something... pushed this change but not showing on github ui..

Screen Shot 2021-03-25 at 4 37 26 PM

Comment (SerKnight) [reply xero_accounting.yaml:4]: Ohh there it is..:) good catch!

**Review by SidneyAllen — APPROVED** [review]
Looks good thanks @SerKnight

**Pull request merged**